ddsrecorder_participants/src/cpp/recorder/mcap/McapHandler.cpp

```diff
@@ -179,10 +185,9 @@ void McapHandler::add_data(
 
     if (state_ == McapHandlerStateCode::STOPPED)
     {
-        logWarning(
-            DDSRECORDER_MCAP_HANDLER,
-            "Attempting to add sample through a stopped handler, dropping...");
-        return;
+        throw utils::InconsistencyException(
```

Comment:
We chose to throw a exception when adding a sample (or schema) failed. So in SchemaWriter, for example, it is called within a try block. It could also have been done with return codes, but we chose this way.